### PR TITLE
[SPARK-34630][PYTHON][SQL] Added typehint for pyspark.sql.Column.contains

### PR DESCRIPTION
**Pull request by dannymeijer**
### What changes were proposed in this pull request?

This PR implements the missing typehints as per SPARK-34630.

### Why are the changes needed?

To satisfy the aforementioned Jira ticket

### Does this PR introduce _any_ user-facing change?

No, just adding a missing typehint for Project Zen

### How was this patch tested?

No tests needed (just adding a typehint)

### CODE REVIEW

**Review by zero323 — COMMENTED** [review]
Thank you for your proposal @dannymeijer.

I'd probably follow Scala side here and use `Any` here:

```python
def contains(self, item: Any) -> Column: ...
```

‒ while it is not that useful, calls like 

```
>>> spark.range(1).select(lit("it is true").contains(True)).show()
+--------------------------+
|contains(it is true, true)|
+--------------------------+
|                      true|
+--------------------------+
```

or 

```
>>> spark.range(1).select(lit("it is true").contains(True)).show()
+--------------------------+
|contains(it is true, true)|
+--------------------------+
|                      true|
+--------------------------+
```

are technically speaking valid.

Could you add `[PYTHON][SQL]` to the title?

File: python/pyspark/sql/column.pyi [hunk 1]
```diff
@@ -115,3 +115,4 @@ class Column:
     def over(self, window: WindowSpec) -> Column: ...
     def __nonzero__(self) -> None: ...
     def __bool__(self) -> None: ...
+    def contains(self, item: Union[str, Column]) -> Column: ...
```
Comment (HyukjinKwon) [inline python/pyspark/sql/column.pyi:118]: Can we also add the type hint for `__version__`?

Comment (dannymeijer) [reply python/pyspark/sql/column.pyi:118]: Isn't __version__ already there?
https://github.com/apache/spark/blob/512f6258071666c3da07f6e6c6299d653a1cc370/python/pyspark/version.pyi#L19

**Comment by dannymeijer** [comment]
> Thank you for your proposal @dannymeijer.
> 
> I'd probably follow Scala side here and use `Any` here:
> 
> ```python
> def contains(self, item: Any) -> Column: ...
> ```
> 
> ‒ while it is not that useful, calls like
> 
> ```
> >>> spark.range(1).select(lit("it is true").contains(True)).show()
> +--------------------------+
> |contains(it is true, true)|
> +--------------------------+
> |                      true|
> +--------------------------+
> ```
> 
> or
> 
> ```
> >>> spark.range(1).select(lit("it is true").contains(True)).show()
> +--------------------------+
> |contains(it is true, true)|
> +--------------------------+
> |                      true|
> +--------------------------+
> ```
> 
> are technically speaking valid.
> 
> Could you add `[PYTHON][SQL]` to the title?

Fair point! Changed it accordingly.

**Comment by HyukjinKwon** [comment]
ok to test

**Review by zero323 — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
@zero323, would you like to try merging this as a brand new committer :-)? I think we can merge this to `master` branch and backport it to `branch-3.1` too.

**Comment by HyukjinKwon** [comment]
@zero323, would be great if you find some time to take a look :-).

**Comment by zero323** [comment]
Sorry, haven't seen your message @HyukjinKwon. On it.

**Comment by zero323** [comment]
Merged to master and branch-3.1. 

Thank you @dannymeijer and @HyukjinKwon

File: python/pyspark/sql/column.pyi (hunk 1 above)
Comment (Yikun) [reply python/pyspark/sql/column.pyi]: The above typehint in version.pyi is ok to some reference like `pyspark.verion.__version`, but doesn't work on some reference like `pyspark.__version`, such like https://github.com/apache/spark/blob/c06758834e6192b1888b4a885c612a151588b390/python/pyspark/pandas/spark/accessors.py#L884

I submit a PR: https://github.com/apache/spark/pull/32110

**Pull request closed without merging**